File: Day  21/160A - Twins.cpp
#include <iostream>
#include <algorithm>
#include <vector>
#include <numeric>
using namespace std;

int main() {
    int n;
    cin >> n;
    vector<int> coins(n);
    for (int i = 0; i < n; i++) {
        cin >> coins[i];
    }
    sort(coins.rbegin(), coins.rend());
    int total = accumulate(coins.begin(), coins.end(), 0);
    int sum = 0;
    int count = 0;
    for (int i = 0; i < n; i++) {
        sum += coins[i];
        count++;
        if (sum > total - sum) {
            break;
        }
    }
    cout << count << endl;
    return 0;
}

File: Day  21/160A - Twins.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        Integer[] coins = new Integer[n];
        for (int i = 0; i < n; i++) {
            coins[i] = sc.nextInt();
        }
        Arrays.sort(coins, Collections.reverseOrder());
        int total = 0;
        for (int coin : coins) {
            total += coin;
        }
        int sum = 0;
        int count = 0;
        for (int i = 0; i < n; i++) {
            sum += coins[i];
            count++;
            if (sum > total - sum) {
                break;
            }
        }
        System.out.println(count);
    }
}
